proto/buoyant-cloud-api.proto

```diff
@@ -92,11 +92,69 @@ message Event {
   Workload owner = 3;
 }
 
+//
+// Linkerd Messages
+//
+
+// This message represents PEM encoded certificate
+// data. It might be a single certificate, a pool of
+// roots or a chain of certificates.
+message CertData {
+  bytes raw = 1;
+}
+
+// Represents the certificates that the control plane
+// has been configured with.
+message ControlPlaneCerts {
+  // This is the identity issuer certificate chain
+  CertData issuer_crt_chain = 1;
+  // The roots that the control plane has been configured with
+  CertData roots = 2;
+}
+
+// A message that carries one or more pod certificates
+message ProxyCerts {
+  repeated ProxyCert certs = 1;
+
+  message ProxyCert {
```

Comment:
We are not really sending a list of all pods. Just their names, namespaces, workload kinds and workload names. I _think_, this is significantly less than what is in being sent on the workload stream.  


> do we want two parallel APIs that are sending up a list of all pods on the cluster? how do we reconcile if they disagree? also we don't necessarily need to decide this right now.

I think these are separate APIs, serving separate purpose and from neatness perspective, I lean on leaving them separate and not mixing them. I however might be missing something. Is there very good reason to try and interleave them? Wrt to getting out of sync, I think even if this happens, given a bit of time things will converge. For example, say a
`ProxyCerts` message arrives before bcloud has received info from the workload stream for  workload/pod.  The certs message will just be ignored because we do not know about this workload/pod. On next cert update iteration, enough time should have passed that the Workload update/refresh has arrived and we already know about this pod, so we insert the certs data. I think in general any of these races that stem from timing issues will get resolved after a few iterations at most. Correct me if I am wrong. In essence if we see a certs update message we only write to the table, we never delete anything. So the only way to get out of sync if we receive it before we know about the pod. And that should get fixed by itself given enough time.